doc/blog/201907-SkiaSharp-for-wasm-support.md

```diff
@@ -0,0 +1,88 @@
+
+```xml
+    <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+      <skia:SKXamlCanvas x:Name="test" PaintSurface="OnPaintSurface" />
+    </Grid>
+```
+
+And with some code behind:
+
+```csharp
+private void OnPaintSurface(object sender, SKPaintSurfaceEventArgs e)
+{
+    // the the canvas and properties
+    var canvas = e.Surface.Canvas;
+
+    // get the screen density for scaling
+    var display = DisplayInformation.GetForCurrentView();
+    var scale = display.LogicalDpi / 96.0f;
+    var scaledSize = new SKSize(e.Info.Width / scale, e.Info.Height / scale);
+
+    // handle the device screen density
+    canvas.Scale(scale);
+
+    // make sure the canvas is blank
+    canvas.Clear(SKColors.Yellow);
+
+    // draw some text
+    var paint = new SKPaint
+    {
+        Color = SKColors.Black,
+        IsAntialias = true,
+        Style = SKPaintStyle.Fill,
+        TextAlign = SKTextAlign.Center,
+        TextSize = 24
+    };
+
+    var coord = new SKPoint(
+        scaledSize.Width / 2, 
+        (scaledSize.Height + paint.TextSize) / 2);
+
+    canvas.DrawText("SkiaSharp", coord, paint);
+}
+```
+
+You can experiment with [this sample from our samples repository](https://github.com/nventive/Uno.Samples/tree/master/UI/SkiaSharpTest).
+
+## The road ahead
+
+There a still a few things we need to for Skia support to be complete. 
+
+**First**, we needed to fork SkiaSharp to enable WebAssembly support, because of the specificities of the native interop layer, noteable because [`Marshal.GetFunctionPointerForDelegate`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.getfunctionpointerfordelegate?view=netframework-4.8) is not supported yet, and [emscripten's `addFunction`](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html) function pointers feature [needs to be used](https://github.com/unoplatform/Uno.SkiaSharp/blob/uno/binding/SkiaSharp.Wasm/ts/SkiaSharpCanvasKit.ts#L21-L32) for the time being. 
+
+We're [working with the SkiaSharp team](https://github.com/mono/SkiaSharp/issues/876) to add an adjustable interop layer that would enable WebAssembly support to be plugged-in at runtime, allowing proper support for packages such as [`SkiaSharp.Extended`](https://www.nuget.org/packages/SkiaSharp.Extended) to be used without being forked to be supported by WebAssembly.
+
+**Second**, WebAssembly specifics in terms of function exports require a very specific set of pre-defined methods ([declared here](https://github.com/mono/mono/blob/8d80ccc897c678d7bdae645ca8629b0c5cc0b667/mono/mini/m2n-gen.cs#L30) in mono-wasm). Altering the existing Skia C API signatures to match those "known" methods is possible to some extents, but not for all methods. This is why some methods are still missing from the Skia implementation, and need to be adjusted to use structures intead of parameter, and avoid updating the mono runtime to be supported.
```

Comment:
```suggestion
**Second**, WebAssembly specifics in terms of function exports require a very specific set of pre-defined methods ([declared here](https://github.com/mono/mono/blob/8d80ccc897c678d7bdae645ca8629b0c5cc0b667/mono/mini/m2n-gen.cs#L30) in mono-wasm). Altering the existing Skia C API signatures to match those "known" methods is possible to some extent, but not for all methods. This is why some methods are still missing from the Skia implementation, and need to be adjusted to use structures instead of parameters, and avoid updating the Mono runtime to be supported.
```